tests/test_basemap.py

```diff
@@ -91,26 +87,88 @@ def test_pmtiles():
     )
     pmtile_file = Path(f"{rootdir}/../test.pmtiles")
     assert pmtile_file.exists()
-
-    # Test reading as form of validation
+    
     with open(pmtile_file, "rb") as pmtile_file:
         data = pmtile_file.read()
     pmtile = PMTileReader(MemorySource(data))
-
+    
     data_length = pmtile.header().get("tile_data_length", 0)
-    assert data_length > 2000 and data_length < 80000
+    assert 2000 < data_length < 80000, "Data length out of expected range"
     assert len(pmtile.metadata().keys()) == 1
-
+    
     metadata = pmtile.metadata()
     attribution = metadata.get("attribution")
     assert attribution == "© esri"
-
+    
     header = pmtile.header()
     min_zoom = header.get("min_zoom")
-    assert min_zoom == 12
-    max_zoom = header.get("max_zoom")
-    assert max_zoom == 14
-
+    assert min_zoom in range(12, 15), f"Expected zoom levels 12-14, but got {min_zoom}"
+
+
+class TestBoundaryHandlerFactory:
+    
+    #def test_init_with_bytesio(self):
+       # boundary = BytesIO(b'{}')
+        #factory = BoundaryHandlerFactory(boundary)
+    
+    def test_init_with_string(self):
+        boundary = "10,20,30,40"
+        factory = BoundaryHandlerFactory(boundary)
+        assert isinstance(factory.handler, StringBoundaryHandler)
+    
+    def test_get_bounding_box(self):
+        boundary = "10,20,30,40"
+        factory = BoundaryHandlerFactory(boundary)
+        assert factory.get_bounding_box() == (10, 20, 30, 40)
+
+class TestBytesIOBoundaryHandler:
+    
+    def setup_method(self):
```

Comment:
This will actually do nothing in the test file.

All test need to start with `test_xxx` to run with pytest.

Perhaps you were planning to use this function elsewhere in the code?